patches/server/0971-More-native-map-PDC-methods-use-adventure-Key.patch

```diff
@@ -0,0 +1,225 @@
++        //Paper start
++        set(key.toString(), type, value);
++    }
++
++    private <T, Z> void set(String key, PersistentDataType<T, Z> type, Z value) {
++        //Paper end
+         Validate.notNull(key, "The provided key for the custom value was null");
+         Validate.notNull(type, "The provided type for the custom value was null");
+         Validate.notNull(value, "The provided value for the custom value was null");
+ 
+-        this.customDataTags.put(key.toString(), this.registry.wrap(type.getPrimitiveType(), type.toPrimitive(value, adapterContext)));
++        this.customDataTags.put(key, this.registry.wrap(type.getPrimitiveType(), type.toPrimitive(value, adapterContext)));
+     }
+ 
+     @Override
+     public <T, Z> boolean has(NamespacedKey key, PersistentDataType<T, Z> type) {
++        //Paper start
++        return has(key.toString(), type);
++    }
++
++    private <T, Z> boolean has(String key, PersistentDataType<T, Z> type) {
++        //Paper end
+         Validate.notNull(key, "The provided key for the custom value was null");
+         Validate.notNull(type, "The provided type for the custom value was null");
+ 
+-        Tag value = this.customDataTags.get(key.toString());
++        Tag value = this.customDataTags.get(key);
+         if (value == null) {
+             return false;
+         }
+@@ -56,10 +73,16 @@ public class CraftPersistentDataContainer implements PersistentDataContainer {
+ 
+     @Override
+     public <T, Z> Z get(NamespacedKey key, PersistentDataType<T, Z> type) {
++        //Paper start
++        return get(key.toString(), type);
++    }
++
++    private <T, Z> Z get(String key, PersistentDataType<T, Z> type) {
++        //Paper end
+         Validate.notNull(key, "The provided key for the custom value was null");
+         Validate.notNull(type, "The provided type for the custom value was null");
+ 
+-        Tag value = this.customDataTags.get(key.toString());
++        Tag value = this.customDataTags.get(key);
+         if (value == null) {
+             return null;
+         }
+@@ -89,9 +112,15 @@ public class CraftPersistentDataContainer implements PersistentDataContainer {
+ 
+     @Override
+     public void remove(NamespacedKey key) {
++        //Paper start
++        remove(key.toString());
++    }
++
++    private void remove(String key) {
++        //Paper end
+         Validate.notNull(key, "The provided key for the custom value was null");
+ 
+-        this.customDataTags.remove(key.toString());
++        this.customDataTags.remove(key);
+     }
+ 
+     @Override
+@@ -189,5 +218,115 @@ public class CraftPersistentDataContainer implements PersistentDataContainer {
+             this.putAll(compound);
+         }
+     }
++
++    @Override
++    public <T, Z> void setIfAbsent(@NotNull net.kyori.adventure.key.Key key, @NotNull PersistentDataType<T, Z> type, @NotNull Z value) {
++        Z z = this.get(key.toString(), type);
```

Comment:
For adventure keys, I think we prefer the `asString` method instead of Object's toString. Because different implementations of Key could do whatever they want with toString, but asString's contract specifies what it has to return.